core/src/main/java/org/apache/james/core/MailAddress.java

```diff
@@ -305,12 +305,12 @@ public String getLocalPart() {
     }
 
     public String asString() {
-        return localPart + "@" + domain;
+        return localPart + "@" + domain.asString();
```

Comment:
what bothers me is that we don't keep the input formating so `MailAddress.of(data).asString()` is not identity (but should)